### PR TITLE
[FEAT]: Close search list when clicked outside of input element

### PR DESCRIPTION
**Pull request by tumero58**

### CODE REVIEW

File: components/searchInput/SearchInput.tsx [hunk 1]
```diff
@@ -25,26 +36,29 @@ export const SearchInput = ({ texts, value, onChange, data, route }: ISearchInpu
...
-              <Typography sx={resultStyles}>{elem.title}</Typography>
-            </Link>
-          );
-        }) : 
-          <Typography sx={resultStyles}>No results found.</Typography>
-        }
-      </Box>}
+      {isOpen && Array.isArray(data) &&
+        <Box sx={resultsWrapper}>
+          {data.length > 0 ? data.map((elem) => {
```
Comment (hrachocode) [inline components/searchInput/SearchInput.tsx:52]:
```suggestion
          {!!data.length ? data.map((elem) => {
```

File: components/searchInput/SearchInput.tsx [hunk 2]
```diff
@@ -17,6 +17,17 @@ interface ISearchInput {
 }
 
 export const SearchInput = ({ texts, value, onChange, data, route }: ISearchInput) => {
+
+  const [ isOpen, setOpen ] = useState(false);
```
Comment (hrachocode) [inline components/searchInput/SearchInput.tsx:21]: what is the purpose of creating additional variable if we were playing with open/false with resultsWrapper?

File: components/searchInput/SearchInput.tsx [hunk 3]
```diff
@@ -25,26 +36,29 @@ export const SearchInput = ({ texts, value, onChange, data, route }: ISearchInpu
...
         <Box sx={{ margin: "12px", display: "flex" }}>
-          <Image src={searchIcon} alt="icon"/>
+          <Image src={searchIcon} alt="icon" />
         </Box>
         <InputBase
           sx={{ flex: 1 }}
           placeholder={texts.input}
           value={value}
           onChange={onChange}
+          onBlur={handleBlur}
```
Comment (hrachocode) [inline components/searchInput/SearchInput.tsx:46]: I see there is something with blur happening. still not sure what was the requirements

**Review by hrachocode — CHANGES_REQUESTED** [review]

File: components/searchInput/SearchInput.tsx (hunk 2 above)
Comment (tumero58) [reply components/searchInput/SearchInput.tsx:21]: we do not want the search result to clear when user clicks off the input. I can change variable names to isFocused, setFocused to not make it confusing.

Comment (hrachocode) [reply components/searchInput/SearchInput.tsx:21]: yeah then lets rename if its about focus

**Review by hrachocode — CHANGES_REQUESTED** [review]

**Review by hrachocode — APPROVED** [review]

**Pull request merged**